fibers.js

```diff
@@ -0,0 +1,25 @@
+const Future = require('fibers/future');
+const fs = Future.wrap(require('fs'));
+const prependDir = (file) => require('path').join(process.argv[2], file)
+
+const main = () => {
+  Future.task(function() {
+    try {
+      process.stdout.write(
+        fs
+        .readFileFuture(prependDir('index.txt'))
+        .wait()
+        .toString()
+        .replace(/\n$/,'')
+        .split('\n')
+        .map((x) => fs.readFileFuture(prependDir(x)).wait().toString())
+        .join(''))
```

Comment:
I found that fibers get popular again after including into native script core. So I just give them a try in that fibers.js. With your comments I found that wasn't great way to use them.
 But coroutines (e.g. fibers) is yet another way and they aren't coupled with Futures. I'll investigate it a bit, and rewrite it purely.